messaging/src/main/java/org/axonframework/eventhandling/SubscribingEventProcessor.java

```diff
@@ -116,6 +120,21 @@ protected void process(List<? extends EventMessage<?>> eventMessages) {
         }
     }
 
+    @Override
+    public void send(Message<?> message, ScopeDescriptor scopeDescription) throws Exception {
+        if (!(message instanceof EventMessage)) {
+            String exceptionMessage = String.format(
+                    "Something else than an EventMessage was scheduled to be dispatched via "
+                            + "SubscribingEventProcessor [%s].", getName());
+            throw new IllegalArgumentException(exceptionMessage);
+        }
+        if (canResolve(scopeDescription)) {
+            EventMessage<?> eventMessage = (EventMessage<?>) message;
+            DefaultUnitOfWork<EventMessage<?>> uow = new DefaultUnitOfWork<>(eventMessage);
+            processInUnitOfWork(Collections.singletonMap(eventMessage, scopeDescription), uow);
```

Comment:
Are you using a `Map` because you are expecting to have several at some stage or purely to couple the event and the scope with one another? Thus that you'd desire to have a `Tuple` or `Map.Entry`?

Reply:
`processInUnitOfWork` takes a map between an event message and its scope. We need this relation in order to invoke the actual scope (usually deadline) message handler. In TEP we're queuing these messages and process them in a batch. In SEP we're invoking one at a time.